libs/feature/editor/src/lib/models/fields.model.ts

```diff
@@ -1,29 +1,47 @@
 import { FormFieldConfig } from '../components/record-form/form-field'
+import { CatalogRecordKeys } from '@geonetwork-ui/common/domain/model/record'
 
 // Expressions should be enclosed in `${}` to be recognized as such
 // eg. ${dateNow()}
 export type EditorFieldExpression = `$\{${string}}`
 
-export interface EditorFieldConfig {
+export interface EditorField {
   // configuration of the form field used as presentation; optional, nothing shown if not defined
-  formFieldConfig?: FormFieldConfig
+  formFieldConfig: FormFieldConfig
 
   // name of the target field in the record; will not change the record directly if not defined
-  model?: string
+  model?: CatalogRecordKeys
 
   // a hidden field won't show but can still be used to modify the record
   // FIXME: currently this is redundant with an absence of formFieldConfig but necessary for clarity
   hidden?: boolean
 
   // the result of this expression will replace the field value on save
   onSaveProcess?: EditorFieldExpression
+
+  value?: EditorFieldValue
+}
+
+export interface EditorFieldSection {
```

Comment:
I would rather have gone for EditorFieldConfig, EditorSectionConfig, EditorPageConfig and EditorConfig, but you can decide what you prefer.